plugin/trino-sqlserver/src/main/java/io/trino/plugin/sqlserver/SqlServerClient.java

```diff
@@ -1068,18 +1068,23 @@ private static Optional<DataCompression> getTableDataCompression(Handle handle,
                 .flatMap(dataCompression -> Enums.getIfPresent(DataCompression.class, dataCompression).toJavaUtil());
     }
 
-    private static Optional<DataCompression> getTableDataCompressionWithRetries(Handle handle, JdbcTableHandle table)
+    public static <T> RetryPolicy<T> getDeadlockRetryPolicy(int maxAttemptCount)
```

Comment:
I think 
```
retryOnDeadlock(() -> getTableDataCompression(handle, table), String exceptionMessage = "error when getting table compression info for '%s'".format(table)) 
```
would look way better. There is no much sense in having maxAttemptCount as a parameter - it could be hardcoded in method.